src/main/java/org/mskcc/limsrest/service/GetQcReportSamplesTask.java

```diff
@@ -179,6 +180,31 @@ protected void getQcSamples(QcReportSampleList rsl, String dataType) throws Exce
         }
     }
 
+    // COVID19 samples can be linked to a request by their OtherSampleId being the concatenation of investigatorId and iLabServiceId
+    protected List<CovidSample> getCovidSamples(String dataType) throws Exception {
+        List<CovidSample> covidSamples = new ArrayList<>();
+        DataRecord requestRecord = dataRecordManager.queryDataRecords("Request", "RequestId = '" + requestId + "'", this.user).get(0);
+
+        String serviceId = (String) requestRecord.getDataField("IlabRequest", this.user);
+
+        List<DataRecord> reportSamplesByRequest = dataRecordManager.queryDataRecords(dataType, "OtherSampleId", otherSampleIds, this.user);
+
+        CovidSample covidSample;
+        for (DataRecord sampleRecord : reportSamplesByRequest) {
+            try {
+                Map<String, Object> sampleFields = sampleRecord.getFields(user);
+                    covidSample = new CovidSample(sampleFields, serviceId);
+                    covidSamples.add(covidSample);
+            } catch (Throwable e) {
```

Comment:
what throws an exception here?

Reply:
Actually, this would be thrown if something went wrong translating lims sample fields to QcReportSamples. So I'd like to keep the Throwable for now and implement a coherent change for these with the next SampleQC release.

Reply:
Good idea